tests/lib/rules/no-pause-test.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * @fileoverview Disallow usage of pauseTest() in tests
+ * @author Scott Kidder
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+let rule = require('../../../lib/rules/no-pause-test'),
+  RuleTester = require('eslint').RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const TEST_FILE_NAME = 'some-test.js';
+
+let ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('no-pause-test', rule, {
+  valid: [
```

Comment:
I would copy the test cases from `no-test-and-then` and just use those since they are simpler and more complete.

Specifically, several test cases are missing:

* `pauseTest()` allowed in non-test file
* `pauseTest.otherFunction()` allowed
* `OtherClass.pauseTest()` allowed